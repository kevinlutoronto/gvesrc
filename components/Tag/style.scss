@import '../../styles/base';
.tag {
  display: inline-block;
  border: 1px solid $cPurple200;
  border-radius: 2px;
  font-weight: bold;
  color: $cPurple200;
  background-color: inherit;

  span {
    white-space: nowrap;
  }
}

/* size */
.small {
  font-size: 11px;
  padding: 3px 10px 2px;
}
