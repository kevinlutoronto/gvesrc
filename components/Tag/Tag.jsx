// @flow
import React from 'react';
import cn from 'classnames';
import { getDataAttrs } from '../../dataUtils';
import type { Data } from '../../types';
import style from './style.scss';

type Props = {|
  text: React$Node,
  children?: React$Node,
  data?: Data,
  className?: string,
|};

const Tag = ({ text, children, data, className = '' }: Props) => {
  const tagClasses = cn(style.tag, style.small, className);
  return (
    <div className={tagClasses} {...getDataAttrs(data)}>
      <span>{text || children}</span>
    </div>
  );
};

Tag.displayName = 'Plasma@Tag';

export default Tag;
