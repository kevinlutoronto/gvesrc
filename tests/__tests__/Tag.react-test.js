/* eslint-disable */

import React from 'react';
import Tag from '../../components/Tag/Tag';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Tag Component', () => {
  test('render', () => {
    const component = renderer.create(<Tag>YOLO</Tag>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
