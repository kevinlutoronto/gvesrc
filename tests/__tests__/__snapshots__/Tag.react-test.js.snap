// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component render 1`] = `
<div
  className="tag small"
>
  <span>
    YOLO
  </span>
</div>
`;
